AnyType/Project_2/AnyType.cpp

```diff
@@ -0,0 +1,160 @@
+#include"DataStructures.h"
+#include"AnyType.h"
+#include<iostream>
+
+
+
+AnyType::AnyType()// empty constructor
+{
+	ValueType = Types::UNKNOWN_TYPE;
+}
+
+AnyType::~AnyType() = default;
+
+AnyType::AnyType(const AnyType &obj)//copy constructor
+{
+	data = obj.data;
+	ValueType = obj.ValueType;
+}
+
+AnyType::AnyType(AnyType &&obj)//move constructor
+{
+	data = obj.data;
+	ValueType = obj.ValueType;
+}
+
+char AnyType::ToChar()
+{
+	if (ValueType != Types::CHAR)// check if current value type != char
+		throw AnyType::BadCastException();//if it's not char throw bad cast exception
+	return data.charData;//if it's char - return char value
+}
+
+int AnyType::ToInt()
+{
+	if (ValueType != Types::INT)
+		throw AnyType::BadCastException();
+	return data.intData;
+}
+
+double AnyType::ToDouble()
+{
+	if (ValueType != Types::DOUBLE)
+		throw AnyType::BadCastException();
+	return data.doubleData;
+}
+
+bool AnyType::ToBool()
+{
+	if (ValueType != Types::BOOL)
+		throw AnyType::BadCastException();
+	return data.boolData;
+}
+
+float AnyType::ToFloat()
+{
+	if (ValueType != Types::FLOAT)
+		throw AnyType::BadCastException();
+	return (data.floatData);
+}
+
+void AnyType::operator=(AnyType const & obj)//copy assigment
+{
+	if (this != &obj)
+	{
+		data = obj.data;
+		ValueType = obj.ValueType;
+	}
+}
+
+void AnyType::operator=(AnyType &&obj)//move assigment
```

Comment:
It might have sense to ensure that value cannot be devires from moved object after it have been moved